### PR TITLE
Initial commit.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	"bufio"
+	"fmt"
+	"path"
+	"log"
+	"os"
+
+	readline "github.com/shylock-hg/readline"
+)
+
+const ttyColorPrefix = "\033["
```
Comment (darionyaphet) [inline cli.go:20]: IMO, you can reference the color package such as [gookit/color](https://github.com/gookit/color)

File: demo.nGQL [hunk 2]
```diff
@@ -0,0 +1 @@
+exit
```
Comment (darionyaphet) [inline demo.nGQL:1]: why need this file?

File: table.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
```
Comment (darionyaphet) [inline table.go:5]: maybe you need [jedib0t/go-pretty](https://github.com/jedib0t/go-pretty)

File: cli.go (hunk 1 above)
Comment (Shylock-Hg) [reply cli.go:20]: Don't plan to use third-party package if no necessary.

File: demo.nGQL (hunk 2 above)
Comment (Shylock-Hg) [reply demo.nGQL:1]: Test script file.

File: table.go (hunk 3 above)
Comment (Shylock-Hg) [reply table.go:5]: Ditto.

**Comment by CPWstatic** [comment]
```
(user@Nebula-Console) [(test1)]> get subgraph 1 steps from "1"
=
|
=
Got 0 rows, 0 columns.
time spent 1205/2806 us
2020-06-12 14:17:56
```
Need a better format for the empty dataset.

**Comment by Shylock-Hg** [comment]
> ```
> (user@Nebula-Console) [(test1)]> get subgraph 1 steps from "1"
> =
> |
> =
> Got 0 rows, 0 columns.
> time spent 1205/2806 us
> 2020-06-12 14:17:56
> ```
> 
> Need a better format for the empty dataset.

Thanks. Good catch.

**Comment by CPWstatic** [comment]
```
(user@Nebula-Console) [(test)]> go from "1" over like yield $^.person.name as name, like._dst as id, like.likeness as likeness
===========================================
|  name       |  id          |  likeness  |
===========================================
|  "william"  |  "2"  |  99        |
-------------------------------------------
```

**Comment by CPWstatic** [comment]
```
(user@Nebula-Console) [(test)]> get subgraph 1 step from "1"
===========================================
|  _vertices  |  _edges                   |
===========================================
|  [1,]       |  [1-[like]->2@0,]  |
-------------------------------------------
```

**Comment by Shylock-Hg** [comment]
> ```
> (user@Nebula-Console) [(test)]> get subgraph 1 step from "1"
> ===========================================
> |  _vertices  |  _edges                   |
> ===========================================
> |  [1,]       |  [1-[like]->2@0,]  |
> -------------------------------------------
> ```

Try the latest commit, I can't simply reproduce this.

**Comment by CPWstatic** [comment]
```
(user@Nebula-Console) [(test)]> get subgraph 6 steps from "1"
============================================================================================================================================================================================================================================================
|  _vertices                         |  _edges                                                                                                                                                                                                             |
============================================================================================================================================================================================================================================================
|  [(1) person.name:"william"]       |  [1-[like]->2@0 likeness:99]                                                                                                                                                                                        |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|  [(2) person.name:"arthur"]        |  [2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99]                                                                                                                                                              |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|  [(1) person.name:"william",(3) ]  |  [1-[like]->2@0 likeness:99,3-[like]->2@0 likeness:99]                                                                                                                                                              |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|  [(2) person.name:"arthur"]        |  [2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99,2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99]                                                                                                          |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|  [(1) person.name:"william",(3) ]  |  [1-[like]->2@0 likeness:99,1-[like]->2@0 likeness:99,3-[like]->2@0 likeness:99,3-[like]->2@0 likeness:99]                                                                                                          |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|  [(2) person.name:"arthur"]        |  [2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99,2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99,2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99,2-[like]->1@0 likeness:99,2-[like]->3@0 likeness:99]  |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

File: main.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	}
+	return false
+}
+
+var t = NewTable(2, "=", "-", "|")
+
+func printResp(resp *graph.ExecutionResponse, duration time.Duration) {
+	// Error
+	if resp.GetErrorCode() != graph.ErrorCode_SUCCEEDED {
+		fmt.Printf("[ERROR (%d)]", resp.GetErrorCode())
```
Comment (CPWstatic) [inline main.go:55]: print message.

Comment (Shylock-Hg) [reply main.go]: Fixed.

File: table.go (hunk 3 above)
Comment (yixinglu) [reply table.go:5]: agree with @darionyaphet 

If so, you can avoid these alignment issues

File: cli.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+
+	readline "github.com/shylock-hg/readline"
```
Comment (yixinglu) [inline cli.go:17]: If using your repo, how do you consider to upgrade readline library later?

Comment (Shylock-Hg) [reply cli.go:17]: This is stable. And it's why I don't plan to introduce too many 3rd package which is personal maintained.

**Comment by yixinglu** [comment]
Maybe you can use libedit to replace the readline library

**Pull request merged**